base/util.go

```diff
@@ -1056,3 +1057,70 @@ func Sha1HashString(str string, salt string) string {
 	hashedKey := h.Sum(nil)
 	return fmt.Sprintf("%x", hashedKey)
 }
+
+// InjectJSONProperty takes the given JSON byte slice, marshals val, and inserts under the given key, and returns the result.
+func InjectJSONProperty(b []byte, key string, val interface{}) (new []byte, err error) {
+	bIsJSONObject, bIsEmpty := isJSONObject(b)
+	if !bIsJSONObject {
+		return nil, errors.New("b is not a JSON object")
+	}
+
+	valBytes, err := json.Marshal(val)
+	if err != nil {
+		return nil, err
+	}
+
+	return injectJSONPropertyFromBytes(b, bIsEmpty, key, valBytes), nil
+}
+
+// InjectJSONPropertyFromBytes takes the given JSON byte slice, and inserts val under the given key, and returns the result.
+func InjectJSONPropertyFromBytes(b []byte, key string, val []byte) (new []byte, err error) {
+	bIsJSONObject, bIsEmpty := isJSONObject(b)
+	if !bIsJSONObject {
+		return nil, errors.New("b is not a JSON object")
+	}
+
+	return injectJSONPropertyFromBytes(b, bIsEmpty, key, val), nil
+}
+
+// isJSONObject checks if the given bytes are a JSON object,
+// and also whether it's an empty object or not.
+func isJSONObject(b []byte) (isJSONObject, isEmpty bool) {
+
+	// Check if the byte slice starts with { and ends with }
```

Comment:
Do we need to worry about leading/trailing whitespace here?

Reply:
Yup, turns out we probably do. Am running bytes.TrimSpace() to do the work on the resulting subslice instead.